alphadia/workflow/peptidecentric.py

```diff
@@ -1160,13 +1164,13 @@ def _build_candidate_speclib_flat(
         Dataframe with scored candidates
     """
     # remove decoys
-    psm_df = psm_df[psm_df["decoy"] == 0]
+    # psm_df = psm_df[psm_df["decoy"] == 0]
 
-    for col in ["rt_observed", "mobility_observed", "mz_observed"]:
-        optional_columns += [col] if col in psm_df.columns else []
+    # make copy to avoid modifying the original dataframe
```

Comment:
please remove the mutable arguments..

https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil

I guess that once we removed the "source of all evil", life will finally be great on planet earth .. so this seems to be a unique chance! :-)

Reply:
I used the pattern from the blog,  it's really a dangerous pattern

Reply:
@jalew188 FYI ;-)